.circleci/config.yml

```diff
@@ -0,0 +1,76 @@
+version: 2
+jobs:
+  # Build job
+  build:
+    working_directory: &working_directory
+      ~/infinityworks/engineering
+    parallelism: 1
+    # environment:
+    #   CIRCLE_ARTIFACTS: /tmp/circleci-artifacts
+    #   CIRCLE_TEST_REPORTS: /tmp/circleci-test-results
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - checkout
+      - setup_remote_docker
+      # - run: mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+      - run:
+          # TODO: rewrite this and run Jekyll as a ruby gem (easier said than done)
+          #
+          # "The job and remote docker run in separate environments"
+          # https://circleci.com/docs/2.0/building-docker-images/#separation-of-environments
+          #
+          # to add insult to the injury:
+          # 'docker cp ~/infinityworks/engineering sources:/source'
+          # results in '/source/engineering/' in the volume container
+          name: Building Jekyll website
+          command: |
+            mkdir -p ~/infinityworks/engineering/build
+            chmod 777 ~/infinityworks/engineering/build
+            # Create the volume container
+            docker create -v /source --name sources alpine:3.4 /bin/true
```

Comment:
Is there a reason this extra volume needs creating? Could you not just mount ~/infinityworks/engineering/build as a writeable volume in the Jekyll container then there wouldn't be the need to create this extra container or copy out of the Jekyll container after it had run.

Reply:
Yeah, there is: "The job and remote docker run in separate environments" from https://circleci.com/docs/2.0/building-docker-images/#separation-of-environments